geopandas/tests/test_geom_methods.py

```diff
@@ -672,6 +679,29 @@ def test_centroid_crs_warn(self):
         with pytest.warns(UserWarning, match="Geometry is in a geographic CRS"):
             self.g4.centroid
 
+    @pytest.mark.skipif(
+        not compat.SHAPELY_GE_18,
+        reason="ignore_index keyword introduced in shapely 1.8.0",
+    )
+    def test_make_valid(self):
+        polygon1 = Polygon([(0, 0), (0, 2), (1, 1), (2, 2), (2, 0), (1, 1), (0, 0)])
+        polygon2 = Polygon([(0, 2), (0, 1), (2, 0), (0, 0), (0, 2)])
+        linestring = LineString([(0, 0), (1, 1), (1, 0)])
+        testGeoseries = GeoSeries([polygon1, polygon2, linestring])
+        out_polygon1 = MultiPolygon(
+            [
+                Polygon([(1, 1), (0, 0), (0, 2), (1, 1)]),
+                Polygon([(2, 0), (1, 1), (2, 2), (2, 0)]),
+            ]
+        )
+        out_polygon2 = GeometryCollection(
+            [Polygon([(2, 0), (0, 0), (0, 1), (2, 0)]), LineString([(0, 2), (0, 1)])]
+        )
+        output = GeoSeries([out_polygon1, out_polygon2, linestring])
+        testGeoseries.is_valid
+        assert_geoseries_equal(testGeoseries.make_valid(), output)
+        testGeoseries.make_valid().is_valid
+
```

Comment:
Can you also add a test checking the error raised for old shapely? Flipping the skipif condition and using `with pytest.raises` to check the error?

Reply:
sorry I'm not sure I understand the request

Reply:
Something like this:

```
@pytest.mark.skipif(
    compat.SHAPELY_GE_18,
    reason="make_valid keyword introduced in shapely 1.8.0",
)
def test_make_valid_shapely_pre18(self):
    s = GeoSeries([Point(1, 1)])
    with pytest.raises(NotImplementedError, match="..."):
        s.make_valid()
```

So we test the part of the code that raises an error if you have an older Shapely version.